### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by X-Elie-X**
### In this PR I implemented  the following:
- Create a class Nameable
-  update Person.rb
- Prepare base Decorator
- Prepare CapitalizeDecorator and TrimmerDecorator
- decorators in action

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require './nameable'
+class Base < Nameable
+  def initialize(nameable)
+    super()
```
Comment (zieeco) [inline base_decorator.rb]: - In this case, the super is redundant and not used so this might be removed

File: base_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require './nameable'
+class Base < Nameable
+  def initialize(nameable)
```
Comment (zieeco) [inline base_decorator.rb]: - Let's add the `attr_accessor :nameable` here as this is needed in the decorator

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @X-Elie-X ,


Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice with the class decorator. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
Hi @X-Elie-X, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**